pkg/provider/kubernetes/preset.go

```diff
@@ -195,12 +195,37 @@ func filterOutPresets(userInfo *provider.UserInfo, list *kubermaticv1.PresetList
 	var presetList []kubermaticv1.Preset
 
 	for _, preset := range list.Items {
+		// find preset based on domains and emails
+		// Example: [ "example.com", "foobar@example.com", "foo.bar@test.com"]
+		// --> all "example.com" emails and "foo.bar@test.com" will get the preset
+		requiredEmails := preset.Spec.RequiredEmails
+
+		// assure backwards compatibility
 		requiredEmailDomain := preset.Spec.RequiredEmailDomain
-		// find preset for specific email domain
 		if requiredEmailDomain != "" {
+			requiredEmails = append(requiredEmails, requiredEmailDomain)
+		}
+
+		if len(requiredEmails) != 0 {
 			userDomain := strings.Split(userInfo.Email, "@")
-			if len(userDomain) == 2 && strings.EqualFold(userDomain[1], requiredEmailDomain) {
-				presetList = append(presetList, preset)
+			for _, emailItem := range requiredEmails {
+				domain := strings.Split(emailItem, "@")
+
+				// if it's a domain, we compare it against the userDomain,
+				// otherwise, it has to match the whole email
+				if len(domain) == 1 {
+					// domain provided
+					if len(userDomain) == 2 && strings.EqualFold(userDomain[1], domain[0]) {
```

Comment:
Did you know that email can contain more then 1 @ symbol?

Reply:
https://www.netmeister.org/blog/email.html

Reply:
Yes, I knew but you don't allow it anyway.
See: https://github.com/kubermatic/kubermatic/blob/master/pkg/provider/kubernetes/preset.go#L201
here, the current code is doing the exact same thing, except that it only allows domains. I don't see a reason why the current code is ok, but the new one gets blocked by this argument.
i.e.:
```
userDomain := strings.Split(userInfo.Email, "@")
if len(userDomain) == 2 && strings.EqualFold(userDomain[1], requiredEmailDomain) {
```
if a user has email `foo@bar@acme.com`. The current code is looking for `bar`!? The new code would allow you to set 
`requiredEmails: ["foo@bar@acme.com"]` and everybody would be happy.